src/catkin_pkg/package.py

```diff
@@ -503,22 +503,23 @@ def parse_package_string(data, filename=None, warnings=None):
     try:
         return _parse_package_string(data, filename=filename, warnings=warnings)
     except InvalidPackage as e:
-        e.args = ('Invalid package manifest "{0}": {1}'.format(filename, e),)
+        if not e.package_path:
+            e.package_path = filename
```

Comment:
In which case would the raise exception not have a package path set?

Reply:
[_get_node() and _get_optional_node()](https://github.com/Nickolaim/catkin_pkg/blob/f63344620230ce72d45d84ade2cbf09a299cf886/src/catkin_pkg/package.py#L686-L697) that e.g. called from [L#534](https://github.com/Nickolaim/catkin_pkg/blob/f63344620230ce72d45d84ade2cbf09a299cf886/src/catkin_pkg/package.py#L534) and [#L616](https://github.com/Nickolaim/catkin_pkg/blob/f63344620230ce72d45d84ade2cbf09a299cf886/src/catkin_pkg/package.py#L616)

Reply:
Can you update the patch to catch, update and raise the exception around those two calls. Then there is no reason for the handling here and `_parse_package_string` could be eliminated all together and the implementation can move back into `parse_package_string` directly.

Reply:
Done.